packages/plugins/Filter/src/types.ts

```diff
@@ -0,0 +1,58 @@
+import { FilterConfiguration } from '@polar/lib-custom-types'
+import Vue from 'vue'
+
+export type LayerId = string
+export type TargetProperty = string
+export type KnownCategory = string
+export type DatePattern = string
```

Comment:
string would suffice technically, but the constructed type isn't very readable when everything is just `Record<string, ...>`. I'd have to add annotations anyway, so I did it in code to prevent something like `string // this is a layerId`.

Reply:
![well](https://github.com/Dataport/polar/assets/73653210/10cf41cc-d02b-4b90-a046-0cc44119b89c)

With primitives, it just resolves back to the above when looking at it at a place, where it is used, loosing all additional information added through the mentioned types.

I think annotations are the way here 👍

Reply:
I disagree. All information is already there, and annotations don't help with the screenshotted situation either.